code/complex_calc/model/complex_num.py

```diff
@@ -0,0 +1,73 @@
+import math
+
+
+class ComplexNum:
+    def __init__(self, re=0.0, im=0.0):
+        self.re = re
+        self.im = im
+
+    def __str__(self):
+        return '{} + {}i'.format(self.re, self.im)
+
+    def __mul__(self, other):
+        if isinstance(other, ComplexNum):
+            return ComplexNum(self.re*other.re - self.im*other.im, self.re*other.im + self.im*other.re)
+        elif isinstance(other, float) or isinstance(other, int):
+            return ComplexNum(self.re * other, self.im * other)
+
+    def __rmul__(self, other):
+        return self.__mul__(other)
+
+    def __add__(self, other):
+        if isinstance(other, ComplexNum):
+            return ComplexNum(self.re + other.re, self.im + other.im)
+        elif isinstance(other, float) or isinstance(other, int):
+            return ComplexNum(self.re + other, self.im)
+
+    def __radd__(self, other):
+        return self.__add__(other)
+
+    def __sub__(self, other):
```

Comment:
`__sub__` should be implemented as` __neg__` and `__add__` composition

Reply:
@xnd-r Ok, reimplemented!